test/testSend.js

```diff
@@ -76,4 +86,35 @@ describe('sql loading and processing', () => {
     const ACTUAL = getExtraParameters(fakeQuery);
     assert.deepEqual(EXPECTED, ACTUAL);
   });
+
+  it('query substitution works', () => {
+    const query = 'SELECT ^something1, ^something2 WHERE ^tablename';
+    const EXPECTED = 'SELECT `Loves`, `Lucy` WHERE `Helix`';
+
+    const params = {
+      tablename: 'Helix',
+      something1: 'Loves',
+      something2: 'Lucy',
+    };
+
+    const ACTUAL = queryReplace(query, params);
+
+    assert.equal(ACTUAL, EXPECTED);
+  });
+
+  it('prevents sql injection by canceling double,single,or template strings', () => {
```

Comment:
Too many commas no ? 
What are "double, single strings ?"

Simple tests like this one are a really good start!
This test shows that it has been written by the developer who coded the logic, which is good to validate your code.
But are you 100% sure that `cancelling double, single or template strings` is enough to prevent SQL injections ? (I do not know, just the theory on tests).
I would add some more tests where you try to be the consumer of your API. Here, some tests which tries to do various real SQL injections might reveal some missing cases. You will not be able to cover all the potential cases but you increase the coverage and reduce the risk.

Reply:
Good point! I will do more to convince users. Query writers should only parameterize Table Names;  cause the tables are all 'requestsYEARMONTH' style. So i'll add an extra check to make sure the string is in style of YEARMONTH.

Ultimately it's up to the query writer too; cause they determine what queries exist in the service as endpoints so they determine parameters, and you can't even send a request without proper Google Credentials and Fastly credentials. Assuming you get through all of that; and still want to be malicious that'd be surprising. So there are a few good safeguards.